spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -557,8 +557,8 @@ public void testBinPackCombineMixedFiles() {
 
   @Test
   public void testBinPackCombineMediumFiles() {
-    Table table = createTable(4);
-    shouldHaveFiles(table, 4);
+    Table table = createTable(6);
+    shouldHaveFiles(table, 6);
```

Comment:
Due to small change from 4 -> 3, it generates 4 taskGroups with last group 1 task. result in 4 files created.
Using 6 -> 3 can have better tolerance.

Reply:
I'm not sure I understand why changes to this test are being made

Reply:
4/3 may produce groups with [1.4, 1.3, 1.3] or [1.3, 1.3, 1.3, 0.1]
The testing data generates  [197078, 188542, 189047, 15617] with small tail. 
Using 4/2 or 6/3 can make test more robust.